enlighten/scope/GainDBFeature.py

```diff
@@ -121,31 +123,39 @@ def reset(self, hotplug=False):
         self.spinbox.setMaximum(self.MAX_GAIN_DB)
         self.spinbox.blockSignals(False)
 
-        # apply the "real" value to spinbox.  This will clamp to supported limits, 
-        # update slider to clamped and send clamped downstream
-        self.spinbox.setValue(now_db)
+        # load gain_db from .ini, falling back to spec.settings.state (copied from EEPROM)
+        ini_gain_db = self.ctl.config.get_float("detector", "gain_db", now_db)
+        log.debug("Get gain from INI: %s", ini_gain_db)
+        self.set_db(ini_gain_db)
 
         log.info("spinbox limits (%d, %d) (current %d)",
             self.spinbox.minimum(), self.spinbox.maximum(), self.spinbox.value())
 
+    def _quiet_set(self, widget, value):
+        """
+        Set the value of a widget without invoking the ValueChanged event
+        """
+        widget.blockSignals(True)
+        widget.setValue(value)
+        widget.blockSignals(False)
+
+
     def set_db(self, db):
-        self.spinbox.setValue(db)
+        
+        # send gain update message to device
+        self.multispec.set_state("gain_db", db)
+        self.multispec.change_device_setting("detector_gain", db)
+
+        # ensure both gain widgets are correct, without generating additional events
+        self._quiet_set(self.spinbox, db)
+        self._quiet_set(self.slider, db)
+
+        # save gain_db to .ini
+        log.debug("Save gain to INI: %s", db)
+        self.ctl.config.set("detector", "gain_db", float(db))
```

Comment:
The problem is, this would force every SiG to use the same gain.  Normally we store and load spectrometer settings on a per-serial number basis.

Settings are normally saved in Configuration.save_file, and loaded (this could arguably move to Spectrometer) in Controller.set_from_ini_file().

Reply:
> Normally we store and load spectrometer settings on a per-serial number basis.

Does this apply to #156, #158, #164 ?

Reply:
Good question.  I would say no, and here's my thinking:

- integration time and gain are very much "device settings" — they're set within the spectrometer, they have "startup" fields in the EEPROM to initialize their per-device settings, etc.  Any setting within ENLIGHTEN which more-or-less directly maps to a hardware setting or existing EEPROM field should probably be tracked per-device.

- plugins (#158) and the cursor (#156 and #164) feel, to me, more like "ENLIGHTEN settings." I think it's fair to treat those as application-level settings.  That said, this could be debated.

To me, the .ini format itself somewhat leans toward "simple settings" (integration time, gain, anything which needs to store exactly one scalar) being "convenient" for per-serial number store, while "complicated settings" where you have to store more than one value, and maybe include sets of values are simpler if given their own Configuration section.

Basically, I'm ambivalent on this.  I personally think it would be easier and more intuitive if Cursor and Plugins were application-wide, but if someone else had a strong opinion otherwise I could be persuaded.

Reply:
We should probably also persist Scan Averaging, and your question remains pertinent: should that be per-serial number or "application-wide?"  In this case I'm going to go with per-serial number, because on at least some spectrometers (SiG), that _will_ be an attribute written to the firmware via USB/BLE.  (And currently it's handled within WasatchDevice, again a per-unit association.)